frontend/src/Editor/DataSourceManager/DataSourceManager.jsx

```diff
@@ -185,14 +185,15 @@ class DataSourceManagerComponent extends React.Component {
             environment_id: currentEnvironment,
           })
           .then(() => {
+            this.props.updateSelecteDatasource(selectedDataSource.name);
             this.setState({ isSaving: false });
             this.hideModal();
             toast.success(
               this.props.t('editor.queryManager.dataSourceManager.toast.success.dataSourceSaved', 'Datasource Saved'),
               { position: 'top-center' }
             );
             this.props.dataSourcesChanged(false, selectedDataSource);
-            this.props.globalDataSourcesChanged();
+            this.props.globalDataSourcesChange && this.props.globalDataSourcesChanged();
```

Comment:
@stepinfwd Don't see the `globalDataSourcesChange` props is being passed. Can you check?